cellmlmanip/_singularity_fixes.py

```diff
@@ -0,0 +1,248 @@
+"""
+``remove_fixable_singularities`` specifies a method to remove fixable singularities from a model
+This represents a flattened CellML model and metadata about its variables.
+"""
+from math import isclose
+
+import networkx as nx
+from sympy import (
+    Abs,
+    Add,
+    Eq,
+    Float,
+    I,
+    Mul,
+    Piecewise,
+    Pow,
+    Wild,
+    exp,
+    log,
+    solveset,
+)
+
+import cellmlmanip
+from cellmlmanip.model import FLOAT_PRECISION
+
+
+ONE = cellmlmanip.Quantity(1.0, 'dimensionless')
+
+
+def _generate_piecewise(expr, V, sp, Vmin, Vmax):
```

Comment:
it can't be internal as it's also used in _remove_singularities. I've tried to explain it better:
```
sp - the value of the singularity point: the value of V for whih U == 0
Vmin - the value of the lower bound: the value of V for which U - U_offset == 0
Vmax - the value of the upper bound: the value of V for which U + U_offset == 0
```
You can't just add 1e-7 to sp, this is something we did try but in some cases the range is too small and in other's it's too big (as in the interpolation doesn't match at the edges) when pickung the value of V for which U +/- 1e-7 ==0 we get a good (non-dimensionalised) range.
It would be possible to pass around U instead, but I decided not to as I didn't want to spread the solving of equations over different places in the code

Reply:
Thanks!